### PR TITLE
Theme Showcase: Use Theme Tiers

### PR DESCRIPTION
**Pull request by Copons**
Related to https://github.com/Automattic/dotcom-forge/issues/4597
Based on #85044

## Proposed Changes

* Add a `useThemeTier` hook and, as a usage example, insert it into the `ThemeTypeBadge` in the Theme Showcase.

| LiTS | LoTS |
|--------|--------|
| Screenshot 2023-12-08 at 16 22 53 | Screenshot 2023-12-08 at 16 21 56 | 

Some notes on the UI changes of this PR:

- LiTS, LoTS, and Design Picker.
- Only the **text** of the labels has been changed; the tooltip has been left out as it's too complex for this PR.
- Only a handful of themes have the new tier taxonomy. For now, everything else is labelled as "free" or "included in my plan".

## Testing Instructions

* Enable the `themes/tiers` feature flag.
* Open the Theme Showcase.
* Check the label of the following themes:
  * Annalee and Hevor: should be limited to the Premium plan.
  * Screenplay: should be limited to the Personal plan.
  * Bookix: should be limited to the Business plan, and require an additional subscription.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by dsas — COMMENTED** [review]
The first thing I realise is that I don't like "Included in my plan", it feels like it should be "Included in your plan" at the least..but maybe just "Included" or something less wordy.

**Comment by dsas** [comment]
Sorry, not really the place but maybe "included in my plan" should eventually have a tool-tip which says something like "This PREMIUM theme is part of your WordPress.com plan" to help reinforce the plan value.

File: client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export default function ThemeTierPartnerBadge( { themeId } ) {
+	const translate = useTranslate();
+	const siteId = useSelector( getSelectedSiteId );
+	const isPartnerThemePurchased = useSelector( ( state ) =>
+		siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false
+	);
+	const { isThemeAllowedOnSite } = useThemeTier( siteId, themeId );
+
+	if ( isPartnerThemePurchased ) {
+		return <span>{ translate( 'Included in my plan' ) }</span>;
```
Comment (dsas) [inline client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:18]: Could this be clearer? It's technically included in your "subscription to this theme plan", but..

Not a blocker for merging, but is something we should address later when we have design input.

**Review by dsas — APPROVED** [review]

**Comment by Copons** [comment]
> The first thing I realise is that I don't like "Included in my plan", it feels like it should be "Included in your plan" at the least..but maybe just "Included" or something less wordy.

I agree with "my/your" comment. It seems that we use a mix of first and second-person already. For example we often use the "Me" and "My Sites", but the existing Premium themes label tooltip uses the "your" form:

Screenshot 2023-12-11 at 14 40 25

@dsas Since this PR is intended to provide rough UI changes behind feature flag, let's move the design discussion to P2: p9Jlb4-a7w-p2#comment-10152

File: client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js (hunk 1 above)
Comment (Copons) [reply client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js:18]: I guess that a "Subscribed" label could be a good fit here.

(But, same recommendation as below about moving design convo to P2: https://github.com/Automattic/wp-calypso/pull/85048#issuecomment-1850221195)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10233640

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Copons for including a screenshot in the description! This is really helpful for our translators.

File: client/components/theme-tier/theme-tier-badge/theme-tier-bundled-badge.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		( state ) => siteId && canUseTheme( state, siteId, themeId )
+	);
+
+	if ( ! bundleSettings ) {
+		return;
+	}
+
+	const BadgeIcon = bundleSettings.iconComponent;
+
+	return (
```
Comment (renatho) [inline client/components/theme-tier/theme-tier-badge/theme-tier-bundled-badge.js:23]: Hi @Copons!

Not sure if it was intentional (it seems it was, based on the logic), so I think it's worth confirming with you :). I noticed that these changes impacted the behavior of the bundled badges.
In the WooCommerce themes, like "Tsubaki", it's displaying 2 badges and not displaying the tooltip in the WooCommerce badge.

Screenshot 2023-12-11 at 18 58 18

And when it's a site with business plan, I also see the WooCommerce badge without the tooltip.

Screenshot 2023-12-11 at 19 18 54

Comment (Copons) [reply client/components/theme-tier/theme-tier-badge/theme-tier-bundled-badge.js:23]: @renatho Yes, that's the expected behaviour, **as long as it's behind feature flag**! 😅 

The UI of the new Tier system is not final, though, and won't for a few more weeks at least. You can chime in on P2: p9Jlb4-a7w-p2#comment-10152 🙂

Comment (renatho) [reply client/components/theme-tier/theme-tier-badge/theme-tier-bundled-badge.js:23]: Ah, perfect! =)
Thank you for the answer and for the reference, @Copons!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10233640).

**Pull request merged**